src/cnxml/serialize.js

```diff
@@ -106,7 +113,9 @@ const MARK_TAGS = {
  * Serializer for figures.
  */
 function figure(obj, children) {
-    return <figure id={obj.key}>
+    let attrs = obj.data.get('class') ? {class: obj.data.get('class') } : {}
```

Comment:
Is there any reason why you are using `obj.data.toJS()` for some nodes, and manually constructing it for others? Also, `obj.data.has()`.

Reply:
`obj.data.toJS()` is passing all arguments, even empty ones and I didn't like that, but probably better way of handling this would be `<figure id={obj.key} ${...removeEmptyKeys(obj.data.toJS())}>`
Would you like me to implement this solution?